apps/dashboard/src/components/workflow-editor/workflow-canvas.tsx

```diff
@@ -275,9 +277,36 @@ const WorkflowCanvasChild = ({ steps, readOnly }: { steps: Step[]; readOnly?: bo
 };
 
 export const WorkflowCanvas = ({ steps, readOnly }: { steps: Step[]; readOnly?: boolean }) => {
+  const has = useHasPermission();
+  const isReadOnly = readOnly || !has({ permission: PermissionsEnum.WORKFLOW_CREATE });
+
   return (
     <ReactFlowProvider>
-      <WorkflowCanvasChild steps={steps || []} readOnly={readOnly} />
+      <div className="relative h-full w-full">
+        <WorkflowCanvasChild steps={steps || []} readOnly={readOnly} />
```

Comment:
I wonder shouw we pass isReadOnly here instead of readOnly?

Reply:
Oh we have 2 "read only" states now which needs to behave differently:
1. one `readOnly` comes from the workflow template store and show preview when you hover over nodes etc, but doesnt really allow you to interact with the wf in any way
2. the other `isReadOnly` is coming from RBAC, but it should allow interaction + shows the tooltip on top etc

I need to figure something out here

Reply:
Oh I see, maybe just making better terminology here in terms of naming.